### ISSUE TITLE
Improve of namespaces

### ISSUE THREAD

**Issue by wojtpl2** [wojtpl2/ExtendedXmlSerializer#133]
For configuration:
```C#
var config = new ConfigurationContainer();
config.EnableDeferredReferences();
config.Type<Section>().Member(p => p.Id).Name("Identity");
config.Type<Section>().EnableReferences(p => p.Id);
IExtendedXmlSerializer exs = config.Create();
```

I need tree namespaces:
```
    using ExtendedXmlSerializer.Configuration;
    using ExtendedXmlSerializer.ExtensionModel.References;
    using ExtendedXmlSerializer.ExtensionModel.Xml;
```

It is not a bag, but nice to have ;) 
What do you think?

**Comment by Mike-EEE** [comment]
No that is fair feedback.  I did try to group extension methods with their extensions, but for the core/built-in functionality we can consolidate them into one file.

I went ahead and did that.  Now, `IExtendedXmlSerializer` is in `ExtendedXmlSerializer.ExtensionModel.Xml` because at some point we were discussing `ExtendedXmlSerializer.ExtensionModel.Json` and that might still be done in the future.  If you use `var exs = config.Create();` instead, it keeps the 3rd namespace from being used. :)
